Build doesn't work with maven 4

### Describe the bug

Building with Maven 4.0.0-rc-5 fails with
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:3.2.8:sign (sign-artifacts) on project            
  infinispan-query: Execution sign-artifacts of goal org.apache.maven.plugins:maven-gpg-plugin:3.2.8:sign failed:            
  org.apache.maven.api.services.ModelBuilderException: 2 problems were encountered while building the effective model for    
  org.infinispan:infinispan-query:jar:16.3.0-SNAPSHOT                                                                        
  [ERROR]     - [ERROR] 'dependencies.dependency.artifactId' for groupId='org.hibernate.search',                             
  artifactId='${hibernate.search.backend.artifactId}', type='jar' with value '${hibernate.search.backend.artifactId}' does   
  not match a valid coordinate id pattern. @ line 44, column 10                                                              
  [ERROR]     - [ERROR] 'dependencies.dependency.version' for groupId='org.hibernate.search',                                
  artifactId='${hibernate.search.backend.artifactId}', type='jar' is missing. @ line 42, column 7
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

--- a/build/configuration/pom.xml
+++ b/build/configuration/pom.xml
@@ -306,14 +306,12 @@
          </activation>
          <properties>
             <version.lucene>${version.lucene9}</version.lucene>
-            <hibernate.search.backend.artifactId>hibernate-search-backend-lucene</hibernate.search.backend.artifactId>
          </properties>
       </profile>
       <profile>
          <id>lucene10</id>
          <properties>
             <version.lucene>${version.lucene10}</version.lucene>
-            <hibernate.search.backend.artifactId>hibernate-search-backend-lucene-next</hibernate.search.backend.artifactId>
          </properties>
       </profile>
    </profiles>

--- a/query/pom.xml
+++ b/query/pom.xml
@@ -40,11 +40,6 @@
       </dependency>
 
       <dependency>
-         <groupId>org.hibernate.search</groupId>
-         <artifactId>${hibernate.search.backend.artifactId}</artifactId>
-      </dependency>
-
-      <dependency>
          <groupId>org.infinispan</groupId>
          <artifactId>infinispan-testing</artifactId>
          <scope>test</scope>
@@ -272,6 +267,27 @@
    </build>
 
    <profiles>
+      <profile>
+         <id>lucene9</id>
+         <activation>
+            <activeByDefault>true</activeByDefault>
+         </activation>
+         <dependencies>
+            <dependency>
+               <groupId>org.hibernate.search</groupId>
+               <artifactId>hibernate-search-backend-lucene</artifactId>
+            </dependency>
+         </dependencies>
+      </profile>
+      <profile>
+         <id>lucene10</id>
+         <dependencies>
+            <dependency>
+               <groupId>org.hibernate.search</groupId>
+               <artifactId>hibernate-search-backend-lucene-next</artifactId>
+            </dependency>
+         </dependencies>
+      </profile>
       <profile>
          <id>smoke</id>
          <build>

